src/containers/widget/index.js

```diff
@@ -0,0 +1,54 @@
+import { h } from "preact";
+import { useState } from "preact/hooks";
+import { CacheProvider } from "@emotion/react";
+
+import { StyledCache } from "../../helpers/styled-cache";
+import { WidgetWrapper } from "../../helpers/widget-wrapper";
+
+import Modal from "../../components/modal";
+import Button from "../../components/button";
+
+import WelcomeScreen from "../../views/welcome-screen";
+
+const Widget = ({ showModal, screenName, imageSrc, title }) => {
+  const [isModalOpen, setModalIsOpen] = useState(showModal);
+
+  const toggleModalVisibility = () => {
+    setModalIsOpen(!isModalOpen);
+  };
+
+  const renderModalContent = () => {
```

Comment:
is this basically our self-made "router"?

Reply:
I looked at the [preact-router](https://github.com/preactjs/preact-router) and [wouter](https://github.com/molefrog/wouter) why don't we use those to navigate between the screens?

Reply:
Since we don't want to change the route of the page on which user is, we shouldn't use preact-router or wouter packages.

Reply:
with change the "route" you mean the URL and the browser history? 
but the routes not necessarily do that, do they? that's an option.

Reply:
I haven't used it before, but [this](https://github.com/molefrog/wouter#uselocation-hook-working-with-the-history) sounded like the location features needs to be used explicitely